src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/pending_resumption_handler.cc

```diff
@@ -0,0 +1,229 @@
+#include "vehicle_info_plugin/pending_resumption_handler.h"
+#include "application_manager/event_engine/event_observer.h"
+#include "application_manager/resumption/resumption_data_processor.h"
+#include "application_manager/message_helper.h"
+
+namespace vehicle_info_plugin {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "PendingResumptionHandler")
+
+PendingResumptionHandler::PendingResumptionHandler(
+    application_manager::ApplicationManager& application_manager)
+    : application_manager::event_engine::EventObserver(
+          application_manager.event_dispatcher())
+    , application_manager_(application_manager) {}
+
+template <class T, class Container>
+bool Contains(const T& val, const Container& container) {
+  return container.find(val) != container.end();
+}
+
+template <class Key, class Value>
+std::set<Key> EnumerateKeys(std::map<Key, Value>& container) {
+  std::set<std::string> keys;
+
+  std::transform(
+      container.begin(),
+      container.end(),
+      std::inserter(keys, keys.end()),
+      [&](const std::pair<std::string, bool>& pair) { return pair.first; });
+
+  return keys;
+}
+
+bool PendingResumptionHandler::IsResumptionResultSuccessful(
+    std::map<std::string, bool>& subscription_results) {
+  bool result = true;
+  for (auto ivi_status : subscription_results) {
+    if (!ivi_status.second) {
+      result = false;
```

Comment:
@mked-luxoft We can do it without `result`, just return `false` if the condition was fulfilled and `true` after loop

Reply:
@ZhdanovP  fixed in 23e9b6736a69f55e45d495d644b00c6437b37002